frontend/src/components/HorizontalScroller.tsx

```diff
@@ -0,0 +1,105 @@
+import { ReactNode, useEffect, useRef, useState } from 'react'
+import { Button } from '@mui/material'
+import { ChevronLeft, ChevronRight } from '@mui/icons-material'
+import clsx from 'clsx'
+
+import './HorizontalScroller.css'
+
+const DEFAULT_SCROLL_OFFSET = 100
+
+interface Props {
+  children?: ReactNode
+  isEnabled?: boolean
+  className?: string
+  scrollOffset?: number
+  [key: string]: any
+}
+
+/**
+ * When enabled, this component will horizontally scroll its children,
+ * showing a left and right scroll button when appropriate.
+ */
+export function HorizontalScroller({
```

Comment:
This component displays content in a horizontal row, and if the content is wider than the container, it will show left and right scroll buttons as necessary.